stdlib/2and3/wsgiref/validate.pyi

```diff
@@ -1,50 +1,53 @@
 import sys
-from typing import Any
+from typing import Any, Iterable, Iterator, Optional
+
+from wsgiref.types import WSGIApplication, InputStream, ErrorStream
 
 class WSGIWarning(Warning): ...
 
-def validator(application): ...
+def validator(application: WSGIApplication) -> WSGIApplication: ...
 
 class InputWrapper:
-    input = ...  # type: Any
-    def __init__(self, wsgi_input): ...
-    def read(self, *args): ...
+    input: InputStream
+    def __init__(self, wsgi_input: InputStream) -> None: ...
     if sys.version_info < (3,):
-        def readline(self): ...
+        def read(self, size: int = ...) -> str: ...
+        def readline(self) -> str: ...
     else:
-        def readline(self, *args): ...
-    def readlines(self, *args): ...
-    def __iter__(self): ...
-    def close(self): ...
+        def read(self, size: int) -> bytes: ...
+        def readline(self, size: int = ...) -> bytes: ...
+    def readlines(self, hint: int = ...) -> bytes: ...
+    def __iter__(self) -> Iterable[bytes]: ...
+    def close(self) -> None: ...  # no return
 
 class ErrorWrapper:
-    errors = ...  # type: Any
-    def __init__(self, wsgi_errors): ...
-    def write(self, s): ...
-    def flush(self): ...
-    def writelines(self, seq): ...
-    def close(self): ...
+    errors: ErrorStream
+    def __init__(self, wsgi_errors: ErrorStream) -> None: ...
+    def write(self, s: str) -> None: ...
+    def flush(self) -> None: ...
+    def writelines(self, seq: Iterable[str]) -> None: ...
+    def close(self) -> None: ...  # no return
 
 class WriteWrapper:
-    writer = ...  # type: Any
-    def __init__(self, wsgi_writer): ...
-    def __call__(self, s): ...
+    writer: Iterable[bytes]
```

Comment:
This doesn't seem right since the code does `self.writer(s)`. It looks like writer has to be a `Callable[[bytes], Any]`.

Reply:
Good catch. Same problem with `__init__()`. Fixing.